app/models/merchant.rb

```diff
@@ -33,4 +33,15 @@ def items_ready_to_ship
       .where(invoice_items: {status: "packaged"})
       .order('invoices.created_at ASC')
   end
+
+  def calculate_top_items
+    self.items
+        .select('items.*, SUM(invoice_items.quantity * invoice_items.unit_price) AS total_revenue')
+        .joins(:invoice_items)
+        .joins(invoices: :transactions) # Change 'invoice' to 'invoices'
+        .where(transactions: { result: 'success' })
+        .group('items.id')
+        .order('total_revenue DESC')
+        .limit(5)
+  end
```

Comment:
I didn't realize that you could use the `.total_revenue` method on items in your view and tests

Nice work!